### PR TITLE
Wip master

### PR DESCRIPTION
**Pull request by AllenDowney**
@AnnJohn @pletcher @jmburges This is ready for review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+		list.addAll(sorted);
+	}
+
+	private List<T> mergeSortHelper(List<T> list, Comparator<T> comparator) {
+	   // TODO: fill this in!
+	   return null;
+	}
+```
+
+These two methods do the same thing but provide different interfaces.  `mergeSort` takes a list and returns a new list with the same elements sorted in ascending order.  `mergeSortInPlace` is a `void` method that modifies an existing list.
```
Comment (pletcher) [inline README.md:129]: Should `mergeSortHelper` be `mergeSortInPlace`?

**Comment by pletcher** [comment]
This looks great! I really like the detailed explanations of bounded heap and radix sort. I fixed a couple of minor things (mostly image links), and I changed the image links to point to S3 (so that Learn doesn't hit issues with GitHub's image hosting).

Other than that, ready to merge when you are!

File: README.md (hunk 1 above)
Comment (AllenDowney) [inline README.md:129]: Yes.  Thanks!

**Comment by AllenDowney** [comment]
@pletcher I think this is ready to go, but Linter doesn't seem to like the java code blocks.  Is something wrong in the setup?

**Comment by pletcher** [comment]
Yep, master fails and solution works!

**Pull request merged**